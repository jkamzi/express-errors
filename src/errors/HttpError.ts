export default class HttpError extends Error {
  public status: number;

  public reason: string;

  constructor(status: number, message: string, reason?: string) {
    super(message);
    this.status = status;
    this.reason = reason || message;

    this.name = 'HttpError';
  }
}
