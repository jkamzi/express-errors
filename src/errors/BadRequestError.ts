import HttpError from './HttpError';

export default class BadRequestError extends HttpError {
  constructor(reason?: string) {
    super(400, 'Bad Request Error', reason);

    this.name = 'BadRequestError';
  }
}