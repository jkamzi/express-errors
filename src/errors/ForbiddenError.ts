import HttpError from './HttpError';

export default class ForbiddenError extends HttpError {
  constructor(reason?: string) {
    super(403, 'Forbidden Error', reason);

    this.name = 'ForbiddenError';
  }
}
