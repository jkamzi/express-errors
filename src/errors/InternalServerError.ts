import HttpError from './HttpError';

export default class InternalServerError extends HttpError {
  constructor(reason?: string) {
    super(500, 'Internal Server Error', reason);

    this.name = 'InternalServerError';
  }
}