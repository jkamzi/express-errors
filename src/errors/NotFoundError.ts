import HttpError from './HttpError';

export default class NotFoundError extends HttpError {
  constructor(reason?: string) {
    super(404, 'Not Found Error', reason);

    this.name = 'NotFoundError';
  }
}
