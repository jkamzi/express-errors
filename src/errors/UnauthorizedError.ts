import HttpError from './HttpError';

export default class UnauthorizedError extends HttpError {
  constructor(reason?: string) {
    super(401, 'Unauthorized Error', reason);

    this.name = 'UnauthorizedError';
  }
}
