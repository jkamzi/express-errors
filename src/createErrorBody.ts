import { HttpError } from './errors';

import type { ErrorResponse } from './types';

export default function createErrorBody(err: HttpError): ErrorResponse {
  return {
    error: {
      message: err.message,
      reason: err.reason,
      status: err.status,
    },
  };
}
