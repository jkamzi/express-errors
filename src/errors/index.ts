import BadRequestError from './BadRequestError';
import ForbiddenError from './ForbiddenError';
import HttpError from './HttpError';
import InternalServerError from './InternalServerError';
import NotFoundError from './NotFoundError';
import UnauthorizedError from './UnauthorizedError';

export {
  BadRequestError,
  ForbiddenError,
  HttpError,
  InternalServerError,
  NotFoundError,
  UnauthorizedError,
};