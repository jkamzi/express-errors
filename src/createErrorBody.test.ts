import createErrorBody from './createErrorBody';
import { HttpError } from './errors';

describe('createErrorBody', () => {
  it('createErrorBody() returns error object', () => {
    expect(
      createErrorBody(new HttpError(400, 'Bad Request', 'Who knows')),
    ).toMatchSnapshot();
  });
});
