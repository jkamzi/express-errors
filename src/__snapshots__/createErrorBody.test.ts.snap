// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createErrorBody createErrorBody() returns error object 1`] = `
Object {
  "error": Object {
    "message": "Bad Request",
    "reason": "Who knows",
    "status": 400,
  },
}
`;
